src/pfs/server/api_server.go

```diff
@@ -77,11 +77,28 @@ func (a *apiServer) InspectRepo(ctx context.Context, request *pfs.InspectRepoReq
 	a.versionLock.RLock()
 	defer a.versionLock.RUnlock()
 	ctx = versionToContext(a.version, ctx)
-	clientConn, err := a.getClientConn(a.version)
+
+	clientConns, err := a.router.GetAllClientConns(a.version)
 	if err != nil {
 		return nil, err
 	}
-	return pfs.NewInternalAPIClient(clientConn).InspectRepo(ctx, request)
+
+	var repoInfos []*pfs.RepoInfo
+	for _, clientConn := range clientConns {
+		repoInfo, err := pfs.NewInternalAPIClient(clientConn).InspectRepo(ctx, request)
+		if err != nil {
+			return nil, err
+		}
+		repoInfos = append(repoInfos, repoInfo)
+	}
+
+	for _, repoInfo := range pfs.ReduceRepoInfos(repoInfos) {
+		if repoInfo.Repo.Name == request.Repo.Name {
+			 return repoInfo, nil
+		}
+	}
```

Comment:
Since we're inspecting a single repo we expect `pfs.ReduceRepoInfos(repoInfos)` to always be of length 1. Let's check that it is, if so return `[0]` otherwise return an error.